include/Dialects/LinalgTransform/FunctionalTransforms.h

```diff
@@ -0,0 +1,77 @@
+//===-- FunctionalTransforms.h - Functional Linalg Transforms -------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef IREE_LLVM_SANDBOX_DIALECTS_LINALGTRANSFORM_FUNCTIONALTRANSFORMS_H
+#define IREE_LLVM_SANDBOX_DIALECTS_LINALGTRANSFORM_FUNCTIONALTRANSFORMS_H
+
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/PatternMatch.h"
+
+namespace mlir {
+namespace linalg {
+
+class FunctionalPaddingPattern {
+public:
+  FunctionalPaddingPattern(
+      const LinalgPaddingOptions &options = LinalgPaddingOptions())
+      : options(options) {}
+
+  FailureOr<LinalgOp> operator()(Operation *op,
+                                 PatternRewriter &rewriter) const;
+
+private:
+  LinalgPaddingOptions options;
+};
+
+class FunctionalBaseTilingPattern {
+public:
+  FunctionalBaseTilingPattern(
+      const LinalgTilingOptions &options = LinalgTilingOptions())
+      : options(options) {}
+
+  FailureOr<TiledLinalgOp> operator()(Operation *op,
+                                      PatternRewriter &rewriter) const;
+
+private:
+  LinalgTilingOptions options;
+};
+
+class FunctionalGenericTilingPattern : FunctionalBaseTilingPattern {
```

Comment:
There will soon be a single LinalgTilingPattern after https://reviews.llvm.org/D116733 lands.